Selenium.HtmlElements/src/Extensions/WebDriverExtensions.cs

```diff
@@ -13,12 +14,24 @@ namespace HtmlElements.Extensions {
     /// </summary>
     public static class WebDriverExtensions {
 
+        private static List<string> InitialWindowHandls;

```

Comment:
you cant just keep state in static variable. It will be shared across all threads and all browsers